amd_openvx_extensions/amd_rpp/source/tensor/SequenceRearrange.cpp

```diff
@@ -136,8 +135,8 @@ static vx_status VX_CALLBACK processSequenceRearrange(vx_node node, const vx_ref
                 unsigned src_index = data->pNewOrder[dst_index];
                 if (src_index > data->sequenceLength)
                     ERRMSG(VX_ERROR_INVALID_VALUE, "invalid new order value=%d (must be between 0-%d)\n", src_index, data->sequenceLength - 1);
-                auto dst_address = (unsigned char *)data->pDst + dst_sequence_start_address + (dst_index * data->pSrcDesc->strides.nStride);
-                auto src_address = (unsigned char *)data->pSrc + src_sequence_start_address + (src_index * data->pDstDesc->strides.nStride);
+                auto dst_address = static_cast<unsigned char *>(data->pDst) + dst_sequence_start_address + (dst_index * data->pSrcDesc->strides.nStride);
+                auto src_address = static_cast<unsigned char *>(data->pSrc) + src_sequence_start_address + (src_index * data->pDstDesc->strides.nStride);
                 memcpy(dst_address, src_address, data->pSrcDesc->strides.nStride);
```

Comment:
Are we considering the datatype of input and output as well in this memcpy?
Because memcpy size will vary based on the datatype

Reply:
nstride does not include the size of the datatype
it is only h * w * c

Reply:
Currently it works on only unsigned char, Will ensure it in future it supports other data types